### DISCUSSION TITLE
New repository `rust-bitcoin-wallet`

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1226] [Ideas]
Raising the idea for discussion to split out our "wallet layer" into a separate repository as suggested in the discussion on #550.

Repo url: github.com/rust-bitcoin/rust-bitcoin-wallet

Workspace crates:
- bip32 (currently a module in rust-bitcoin)
- bip39 (currently a crate in rust-bitcoin org
- psbt  (currently a module in rust-bitcoin)

As far a I understand the benefit of a separate repository is that the pace of development and the depth/time/people reviewing can be different for different repos so having a wallet repo would allow that repo to progress at a faster pace without risking the crates lower down the stack.

**Comment by tcharding** [comment]
Note "bitcoin-wallet" is already taken on crates.io by our `rust-wallet` crate.

**Comment by dpc** [comment]
As an alternative: you could use `CODEOWNERS` and just agree on some different rules of stability/review for different parts of the codebase.
